assets/js/modules/analytics-4/datastore/report.test.js

```diff
@@ -425,6 +429,28 @@ describe( 'modules/analytics-4 report', () => {
 
 					expect( isGatheringData() ).toBe( false );
 				} );
+
+				it( 'should return TRUE without checking for property settings if the report has zero data and user is not authenticated', async () => {
+					registry.dispatch( CORE_USER ).receiveGetAuthentication( {
```

Comment:
Hey @jimmymadon, as we were discussing writing a test for the case where `getAuthentication()` is not resolved yet, I had a look and was able to come up with something. This could be added within this `describe.each()` set of tests although the call to `receiveGetAuthentication()` would need to be moved out of the `beforeEach()` and into each test that needs it. What do you think?

```js
it( 'should return undefined if getAuthentication is not resolved yet', async () => {
	fetchMock.getOnce(
		new RegExp(
			'^/google-site-kit/v1/core/user/data/authentication'
		),
		{
			authenticated: false,
		}
	);

	const { isGatheringData, hasZeroData } =
		registry.select( MODULES_ANALYTICS_4 );

	// The first call to isGatheringData returns undefined because the call to hasZeroData returns undefined.
	expect( isGatheringData() ).toBeUndefined();
	expect( hasZeroData() ).toBeUndefined();

	// Wait for the next tick to allow getReport to resolve. We don't use waitForDefaultTimeouts as the longer timeout in that function
	// can allow the getAuthenticated call to resolve too, which we don't want here.
	await new Promise( ( resolve ) => {
		setTimeout( resolve, 1 );
	} );

	// The second call to isGatheringData returns undefined because the call to isAuthenticated returns undefined.
	expect( isGatheringData() ).toBeUndefined();
	// This confirms that hasZeroData is now resolved.
	expect( hasZeroData() ).toBe( true );

	await untilResolved(
		registry,
		CORE_USER
	).getAuthentication();

	// Verify that isGatheringData now returns TRUE as the call to getAuthentication has resolved and user is not authenticated.
	expect( isGatheringData() ).toBe( true );
	expect( hasZeroData() ).toBe( true );
} );
```